synapse/federation/federation_client.py

```diff
@@ -533,6 +535,9 @@ async def _try_destination_list(
                 next server tried. Normally the stacktrace is logged but this is
                 suppressed if the exception is an InvalidResponseError.
 
+            failover_errcodes: Error codes (specific to this endpoint) which should
+                cause a failover.
```

Comment:
we should probably document that they only cause a failover when in companionship with a 400 HTTP error.

Reply:
Good call. Thanks!